subsys/net/lib/coap_utils/coap_utils.c

```diff
@@ -188,14 +190,16 @@ static void coap_set_response_callback(struct coap_packet *request,
 	reply->reply = reply_cb;
 }
 
-void coap_init(int ip_family)
+void coap_start(int ip_family)
 {
 	proto_family = ip_family;
 
 	fds.events = POLLIN;
 	fds.revents = 0;
 	fds.fd = coap_open_socket();
 
+	is_active = true;
+
 	/* start sock receive thread */
 	k_thread_create(&receive_thread_data, receive_stack_area,
```

Comment:
This thread should be spawned once (AFAIR it can be done statically with macros) and it should not terminate. Currently, Zephyr does not guarantee that re-spawning a thread which terminated, will actually work:
https://docs.zephyrproject.org/latest/reference/kernel/threads/index.html#thread-termination
I am aware this appears to work (have not investigated in-depth to know if there are any side-effects), but since it's not supported by the kernel, we should rather not abuse this feature. All in all we provide samples/reference code here.

The thread exectution can be controlled with some kernel primitive (like blocking on a semaphore until `coap_start()` is called).

Reply:
I reverted changes in coap_utils, we don't need to change here anything actually.